bin/install

```diff
@@ -34,10 +34,8 @@ install_python() {
   fi
 
   if [ "$install_path" != "$symlink_path" ]; then
-    # add a symlink for an alias like:
-    # ~/.asdf/installs/python/3 -> ~/.asdf/installs/python/3.1.2
-    rmdir "$symlink_path" # asdf creates this empty directory we don't want
-    ln -sfv "$install_path" "$symlink_path"
```

Comment:
I see.  But I feel that we are being asked to pay a heavy price for fuzzy matching.  My workflow tends to be:
```shell
for plugin in "${plugins[@]}"; do 
    latest=$(asdf latest $plugin ${partial[$plugin]})
    asdf install $plugin $latest
    asdf global $plugin $latest
done
```
Conditionally requiring a `-latest` suffix doesn't really help me here.

Reply:
Assuming we can land the bug fix in #141 your current workflow works totally fine:

```
❯ asdf latest python 3.10
3.10.6
❯ asdf install python 3.10.6
Installed Python-3.10.6 to /Users/jdx/.asdf/installs/python/3.10.6
❯ asdf global python 3.10.6
❯ ~/.asdf/shims/python -V
Python 3.10.6
```

The difference is that your workflow could now be simplified:

```
❯ asdf install python 3.10-latest
Installed Python-3.10.6 to /Users/jdx/.asdf/installs/python/3.10.6
/Users/jdx/.asdf/installs/python/3.10-latest -> /Users/jdx/.asdf/installs/python/3.10.6
❯ asdf global python 3.10-latest
❯ ~/.asdf/shims/python -V
Python 3.10.6
```

Yes, #139 has a bug that breaks your current workflow but that wasn't intentional and will be fixed in #141. There's no need to toss the baby out with the bathwater here. I've applied what I think is the necessary portion of this PR to retain POSIX compatibility as well.

Reply:
I agree a polyglot solution would be better so that you could do something similar for all languages. I am considering submitting a PR for that but it's definitely not as easy change and may have performance considerations. asdf is already way too slow IMO